Semantics.md

```diff
@@ -831,29 +831,39 @@ outside the range which rounds to an integer in range) traps.
 
 ## Thread operators
 
-  * `is_lock_free`: Given an operand `N`, if the atomic step of an atomic
-    primitive (see [Atomic Memory Accesses](#atomic-memory-accesses)) on a
-    datum of size `N` bytes will be performed without the agent acquiring a
-    lock outside the `N` bytes comprising the datum, then return `1`.
-    Otherwise, return `0`.
+  * `is_lock_free`: Given an operand `N` of type `i32`, if the atomic step of
+    an atomic primitive (see [Atomic Memory Accesses](#atomic-memory-accesses))
+    on a datum of size `N` bytes will be performed without the [agent](agent)
+    acquiring a lock outside the `N` bytes comprising the datum, then return
+    `1`. Otherwise, return `0`. Once the value of `is_lock_free` for a given
+    value `N` has been observed for any [agent](agent) in an
+    [agent cluster](agent cluster) , it cannot change.
 
 ### Wait
 
 The wake and wait operators are optimizations over busy-waiting for a value to
 change. It is a validation error to use these operators on non-shared linear
-memory.
+memory. They are non-atomic and have sequentially consistent memory ordering.
+
+Both wake and wait operators trap if the effective address of either operator
+is misaligned or out-of-bounds.
 
-The wait operators take three operands: an address operand, an expected
-value, and a relative timeout in milliseconds as an f64. The timeout must be
-non-negative, and can also be positive infinity. If the timeout is positive
-infinity then the timeout will never expire.
+The wait operator take three operands: an address operand, an expected
+value as an `i32`, and a relative timeout in milliseconds as an `f64`.
 
-The wait operation begins by performing an atomic load from the given address.
-If the loaded value is not equal to the expected value, the operator returns 1
-("not-equal"). If the values are equal, the thread is suspended. If the thread
-is woken (by execution of the corresponding "wake" operator on another thread),
-the wait operator returns 0 ("ok"). If the timeout expires before another
-thread wakes this one, this operator returns 2 ("timed-out").
+| `timeout` value | Behavior |
+| ---- | ---- |
+| `timeout` <= 0 | Expires immediately |
```

Comment:
"immediately": is it guaranteed synchronous, or can it wait until the next runloop?